src/client/components/GetMeal.js

```diff
@@ -0,0 +1,129 @@
+import AddReservation from "./AddReservation";
+import React, { useState, useEffect } from "react";
+import { useParams} from 'react-router-dom';
+import MealsImages from "./MealsImages";
+
+const GetMeal = ()=>{
+    const [meal, setMeal] = useState({});
+    const [reservation, setReservation] = useState(false);
+    const [availableReservations, setAvailableReservations] = useState(null)
+    const [totalGuests, setTotalGuests] = useState(null)
+    const [message, setMessage] = useState("");
+    const [error, setError] = useState(null);
+    const [loading, setLoading] = useState(true);
+    
+    const params = useParams();
+    useEffect(() =>{
+        const fetchingMealsApi = async() => {
+            const API_URL = `/api/meals/${params.id}`;
+            
+            try{
+                setLoading(true);
+                setError(null);
+
+                const response = await fetch(API_URL);
+                if(response.ok){
+                    const mealInfo = await response.json();
+                    setMeal(mealInfo[0])
+                } else {
+                    const errorResult = await response.json();
+                    console.log('Error in fetching meal info: ', errorResult.message);
+                    setError(errorResult);
+                }
+                setLoading(false);
+            } catch(error){
+                console.log('Error in fetching meal info: ', error.message);
+                setError(error);
+            }
+        }
+        fetchingMealsApi();
```

Comment:
Try to create some reusable logic for the fetching, error handling and loading as I mentioned above. This requires a bit more thinking